quic/s2n-quic-transport/src/space/mod.rs

```diff
@@ -935,3 +932,13 @@ pub trait PacketSpace<Config: endpoint::Config> {
         Ok(processed_packet)
     }
 }
+
+pub(crate) fn into_pn_range_iter(
+    iter: impl Iterator<Item = RangeInclusive<VarInt>>,
+    space: PacketNumberSpace,
+) -> impl Iterator<Item = PacketNumberRange> {
+    iter.map(move |ack_range| {
+        let (start, end) = ack_range.into_inner();
+        PacketNumberRange::new(space.new_packet_number(start), space.new_packet_number(end))
+    })
+}
```

Comment:
Should we just put this on the `frame::Ack`? Seems useful to have

```rust
impl Ack {
    pub fn pn_ranges(&self, space: PacketNumberSpace) -> impl Iterator<Item = PacketNumberRange> + '_ {
        ...
    }
}
```

Reply:
yea good idea: https://github.com/aws/s2n-quic/pull/1311/commits/c6330d47648c0323eca8995cc8d7cbf659690cf3